packages/vite/src/node/plugins/importMetaGlob.ts

```diff
@@ -594,7 +601,16 @@ export async function toAbsoluteGlob(
   if (glob.startsWith('../')) return pre + posix.join(dir, glob)
   if (glob.startsWith('**')) return pre + glob
 
-  const resolved = normalizePath((await resolveId(glob, importer)) || glob)
+  const isSubImportsPattern = glob.startsWith('#') && glob.includes('*')
+
+  const resolved = normalizePath(
+    (await resolveId(glob, importer, {
+      custom: { 'glob-imports': isSubImportsPattern },
```

Comment:
@bluwy I also think we should merge this one, we have time to fix Astro if it is affecting it (but as you say, I doubt is related)

About the `custom` option, maybe we should use a namespace for it to avoid collisions? Something like `'vite:import-glob:imports'` or `'vite:import-glob-imports'` ?